src/main/java/software/amazon/smithy/lsp/ext/SmithyProject.java

```diff
@@ -190,61 +194,157 @@ private static Map<ShapeId, Location> collectLocations(Model model) {
+                Location containerLocation = locations.get(entry.getKey());
+                Range containerLocationRange = containerLocation.getRange();
+                int memberEndMarker = containerLocationRange.getEnd().getLine();
+                // Keep track of previous line to make sure that end marker has been advanced.
+                String previousLine = "";
+                // The member shapes were reverse ordered by source location when assembling this list, so we can
+                // iterate through it as-is to work from bottom to top in the model file.
+                for (MemberShape memberShape : entry.getValue()) {
+                    int memberShapeSourceLocationLine = memberShape.getSourceLocation().getLine();
+                    // If the member's source location matches the container location's starting line (with offset),
+                    // the member is mixed in and not present in the model file.
+                    if (memberShapeSourceLocationLine == containerLocationRange.getStart().getLine() + 1) {
+                        locations.put(memberShape.getId(), createElidedMemberLocation(containerLocation));
+                    // Otherwise, determine the correct location by trimming comments, empty lines and applied traits.
+                    } else {
+                        String currentLine = lines.get(memberEndMarker - 1).trim();
+                        while (currentLine.startsWith("//") || currentLine.equals("") || currentLine.equals("}")
+                                || currentLine.startsWith("@") || currentLine.equals(previousLine)) {
+                            memberEndMarker = memberEndMarker - 1;
+                            currentLine = lines.get(memberEndMarker - 1).trim();
+                        }
+                        Position startPosition = getStartPosition(memberShape.getSourceLocation());
+                        Position endPosition = getEndPosition(memberEndMarker, lines);
+                        // Advance the member end marker on any non-mixin traits on the current member, so that the next
+                        // member location end is correct. Mixin traits will have been declared outside the
+                        // containing shape and shouldn't impact determining the end location of the next member.
+                        List<Trait> traits = memberShape.getAllTraits().values().stream()
+                                .filter(trait -> !trait.getSourceLocation().equals(SourceLocation.NONE))
+                                .filter(trait -> trait.getSourceLocation().getFilename().equals(modelFile))
+                                .filter(trait -> !isFromMixin(containerLocationRange, trait))
+                                .collect(Collectors.toList());
+                        if (!traits.isEmpty()) {
+                            traits.sort(new SourceLocationSorter());
```

Comment:
```suggestion
                            traits.sort(Comparator.comparing(Trait::getSourceLocation));
```

Reply:
Fixed in [0ed9978](https://github.com/awslabs/smithy-language-server/pull/61/commits/0ed9978ed9b3d58df77fb1af80310514f8d352bd).